test/framework/src/main/java/org/elasticsearch/cluster/routing/TestShardRouting.java

```diff
@@ -287,6 +287,31 @@ public static ShardRouting newShardRouting(
         );
     }
 
+    public static ShardRouting newShardRouting(
```

Comment:
We really should think about getting all these overloads under control :)

Reply:
I have something in mind, going to open a pr with it today or tomorrow

Reply:
https://github.com/elastic/elasticsearch/pull/104472